### PR TITLE
[FIX] mail,hr: open partners avatar card instead of chat thread

### PR DESCRIPTION
**Pull request by maad-odoo**
### Prior to this commit:
Clicking on a partner's `@mention` in the chat interface would directly open 
the corresponding chat thread.

### Following this commit:
Upon clicking a partner's `@mention`, the system now presents an avatar card 
containing pertinent information about the mentioned partner. If the partner 
has a user_id then display the 'Send Message' .

task-3816952

### CODE REVIEW

File: addons/mail/static/tests/message/message.test.js [hunk 1]
```diff
@@ -1380,7 +1380,7 @@ test("data-oe-id & data-oe-model link redirection on click", async () => {
     await assertSteps(["do-action:openFormView_some.model_250"]);
 });
 
-test("Chat with partner should be opened after clicking on their mention", async () => {
+test("Avatar Card of the partner should be opened after clicking on their mention", async () => {
```
Comment (rapr-odoo) [inline addons/mail/static/tests/message/message.test.js:1383]: Should also test when partner doesn't have a dedicated user.

File: addons/mail/static/tests/message/message.test.js [hunk 2]
```diff
@@ -1395,8 +1395,9 @@ test("Chat with partner should be opened after clicking on their mention", async
     await contains(".o-mail-Composer-input", { value: "@Test Partner " });
     await click(".o-mail-Composer-send:enabled");
     await click(".o_mail_redirect");
-    await contains(".o-mail-ChatWindow .o-mail-Thread");
-    await contains(".o-mail-ChatWindow", { text: "Test Partner" });
+    await contains(".o_avatar_card");
+    await contains(".o_card_user_infos", { text: "Test Partner" });
```
Comment (rapr-odoo) [inline addons/mail/static/tests/message/message.test.js:1399]: Could be done in a single `contains`.

**Review by rapr-odoo — CHANGES_REQUESTED** [review]
Looks good! Some changes for naming.

File: addons/mail/controllers/thread.py [hunk 3]
```diff
@@ -127,3 +127,19 @@ def mail_message_update_content(self, message_id, body, attachment_ids, attachme
             message_sudo, body, attachment_ids=attachment_ids, partner_ids=partner_ids
         )
         return message_sudo.message_format()[0]
+
+    @http.route("/mail/message/mention_redirect", methods=["POST"], type="json", auth="public")
+    @add_guest_to_context
```
[the remark this answers is not in the document]
Comment (maad-odoo) [reply addons/mail/controllers/thread.py]: No, the guest will not be able to open the avatar card.

**Comment by maad-odoo** [comment]
@rapr-odoo Changes are done. Please have a look!

File: addons/mail/static/src/discuss/web/avatar_card/avatar_card_popover.xml [hunk 4]
```diff
@@ -31,7 +31,7 @@
                 <div class="flex-wrap gap-2 mt-2">
                     <div class="justify-content-end d-flex align-items-baseline">
                         <div class="d-flex gap-2 o_avatar_card_buttons">
-                            <button t-if="!user.share" class="btn btn-secondary btn-sm" t-on-click.stop="onSendClick">Send message</button>
+                            <button t-if="user.hasOwnProperty('share') &amp;&amp; !user.share" class="btn btn-secondary btn-sm" t-on-click.stop="onSendClick">Send message</button>
```
Comment (rapr-odoo) [inline addons/mail/static/src/discuss/web/avatar_card/avatar_card_popover.xml:34]:
```suggestion
                            <button t-if="!(user.share ?? true)" class="btn btn-secondary btn-sm" t-on-click.stop="onSendClick">Send message</button>
```

**Review by rapr-odoo — CHANGES_REQUESTED** [review]
Some remarks for code optimisation.

**Comment by maad-odoo** [comment]
@rapr-odoo Changes are done. Please have a look.

File: addons/mail/static/tests/message/message.test.js [hunk 5]
```diff
@@ -1415,8 +1415,26 @@ test("Chat with partner should be opened after clicking on their mention", async
     await contains(".o-mail-Composer-input", { value: "@Test Partner " });
     await click(".o-mail-Composer-send:enabled");
     await click(".o_mail_redirect");
-    await contains(".o-mail-ChatWindow .o-mail-Thread");
-    await contains(".o-mail-ChatWindow", { text: "Test Partner" });
+    await contains(".o_avatar_card .o_card_user_infos", { text: "Test Partner" });
+    await contains(".o_card_user_infos", { text: "testpartner@odoo.com" });
+});
+
+test("Avatar Card of the partner should be opened after clicking on their mention without corresponding user", async () => {
```
Comment (rapr-odoo) [inline addons/mail/static/tests/message/message.test.js:1422]: This test case could be done in above test case. So, the title would be something like `Avatar Card of the partner should be opened after clicking on their mention`

File: addons/mail/static/src/core/common/message.js [hunk 6]
```diff
@@ -423,11 +422,9 @@ export class Message extends Component {
         }
         if (ev.target.closest(".o_mail_redirect")) {
             ev.preventDefault();
-            const partnerId = Number(ev.target.dataset.oeId);
-            if (user.partnerId !== partnerId) {
-                this.threadService.openChat({ partnerId });
+            if (this.store.self.type === "guest") {
+                return;
             }
-            return;
```
Comment (rapr-odoo) [inline addons/mail/static/src/core/common/message.js:430]: Only Public and Portal users will be using this code.  Therefore, just return here without any checks, as Public/Portal users won't be able to access the form views of the Partner.

**Review by rapr-odoo — CHANGES_REQUESTED** [review]
Some remarks and nitpicks.

**Review by rapr-odoo — APPROVED** [review]

**Comment by maad-odoo** [comment]
Will be continued in this https://github.com/odoo/odoo/pull/164355 PR

**Pull request closed without merging**